db/python/layers/web.py

```diff
@@ -372,20 +387,33 @@ async def get_project_summary(
         seen_seq_types = set(cram_number_by_seq_type.keys()).union(
             set(seq_number_by_seq_type.keys())
         )
-        sequence_stats = {}
+        seen_batches = set(seq_number_by_seq_type_and_batch.keys())
+
+        sequence_stats: Dict[str, Dict[str, str]] = {}
+        cram_seqr_stats = {}
+
         for seq in seen_seq_types:
-            sequence_stats[seq] = {
+            cram_seqr_stats[seq] = {
                 'Sequences': str(seq_number_by_seq_type.get(seq, 0)),
                 'Crams': str(cram_number_by_seq_type.get(seq, 0)),
                 'Seqr': str(seqr_stats_by_seq_type.get(seq, 0)),
             }
 
+        for batch in seen_batches:
+            sequence_stats[batch] = {}
```

Comment:
This is fine, but also a good use for `defaultdict`

Reply:
I figured we didn't need a defaultdict for this but actually I'll make the code more Pythonic with list comprehension. Then no need to create an empty dict first anyway -
```dict([(seq, seq_number_by_seq_type_and_batch[batch].get(seq, 0)) for seq in seen_seq_types])```